docs/guides/migrating-cra.md

```diff
@@ -0,0 +1,774 @@
+
+If you inject `<link />` tags into your page client-side in your existing route components, either directly or via an abstraction like [`react-helmet`](https://www.npmjs.com/package/react-helmet), you'll get to delete a lot of code and possibly a dependency or two!
+
+### Route `<meta />` tags
+
+Similar to `links`, each route can also export a `meta` function that—you guessed it—returns a value responsible for rendering `<meta />` tags for that route. This is useful because each route often has its own
+
+The API is slightly different for `meta`. Instead of an array, it returns an object where the keys represent the meta `name` attribute (or `property` in the case of OpenGraph tags) and the value is the `content` attribute. The object can also accept a `title` property that renders a `<title />` component specifically for that route.
+
+Again—no more weird dances to get meta into your routes from deep in the component tree. Export them at the route level and let the server handle it. ✨
+
+## Using the `.jsx` extension for React component files
+
+In CRA you can use JSX in either `.jsx` or `.js` files. You'll need to rename any components with JSX to `.jsx`.
+
+## Updating imports and dev scripts
+
+Remix re-exports everything you get from `react-router-dom` and we recommend that you update your imports to get those modules from `@remix-run/react`. In many cases, those components are wrapped with additional functionality and features specifically optimized for Remix.
+
+**Before:**
+
+```jsx bad
+import { Link, Outlet } from "react-router-dom";
+```
+
+**After:**
+
+```jsx
+import { Link, Outlet } from "@remix-run/react";
+```
+
+You should also update a few scripts in your `package.json` to run Remix instead of `react-scripts`:
+
+**After:**
+
+```json
+{
+  "start": "react-scripts start",
+  "dev": "react-scripts start",
+  "build": "react-scripts build"
+}
+```
+
+**After:**
+
+```json
+{
+  "build": "remix build",
+  "start": "cross-env NODE_ENV=production node ./server.js",
+  "dev": "remix build && run-p dev:*",
+  "dev:node": "cross-env NODE_ENV=development nodemon ./server.js --watch ./server.js",
+  "dev:remix": "remix watch"
+}
+```
+
+These scripts are a bit more verbose, but that's because we don't abstract the processes. We give that power back to you. After all, you know how to run your app better than anyone on our team!
+
+## Final Thoughts
+
+While we've done our best to provide a comprehensive migration guide, it's important to note that we built Remix from the ground up with a few key principles that differ significantly from how many React apps are currently built. While your app will likely run at this point, as you dig through our docs and explore our APIs, we think you'll be able to drastically reduce the complexity of your code and improve the end user experience of your app. It might take a bit of time to get there, but you can eat that elephant one bite at a time. 🐘
```

Comment:
```suggestion
While we've done our best to provide a comprehensive migration guide, it's important to note that we built Remix from the ground up with a few key principles that differ significantly from how many React apps are currently built. While your app will likely run at this point, as you dig through our docs and explore our APIs, we think you'll be able to drastically reduce the complexity of your code and improve the end user experience of your app. It might take a bit of time to get there, but you can eat that elephant one bite at a time. 🐘 In particular, after you've shipped this first upgrade to your app, you'll want to start moving your data fetching and mutations to [`loaders`](../api/conventions#loader) and [`actions`](../api/conventions#action). Get ready to delete a lot of code!
```